crates/wysiwyg/src/tests/test_deleting.rs

```diff
@@ -361,3 +361,19 @@ fn deleting_selection_of_a_container_in_multiple_containers() {
     model.backspace();
     assert_eq!(tx(&model), "<i>| test</i>");
 }
+
+#[test]
+fn deleting_selection_of_a_container_with_text_node_neighbors() {
+    let mut model = cm("<em>abc<del>{def}|</del>ghi</em>");
+    model.backspace();
+    assert_eq!(tx(&model), "<em>abc|ghi</em>");
+}
+
+#[test]
+fn deleting_selection_of_a_container_with_matching_neighbors() {
+    let mut model = cm(
+        "<em><strong>abc</strong><del>{def}|</del><strong>ghi</strong></em>",
+    );
+    model.backspace();
+    assert_eq!(tx(&model), "<em><strong>abc|ghi</strong></em>");
```

Comment:
maybe is worth checking the tree to for these two tests?

Reply:
I just added `explicitly_assert_invariants()` to all deleting tests that did not have it before and they all pass. Having this test pass and the expected HTML should be pretty much the same as having the tree in the form we expect :)